src/main/java/lab2/AvatarGenerator.java

```diff
@@ -1,66 +1,100 @@
-package lab2;
-
-import javax.imageio.ImageIO;
-import javax.swing.*;
-import java.awt.*;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URI;
-import java.net.http.HttpClient;
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
+package lab2; // Defines the package for this class, useful for organizing code into namespaces
 
+// Importing necessary Java libraries
+import javax.imageio.ImageIO; // For reading and writing images
+import javax.swing.*; // For creating GUI components like JFrame and JLabel
+import java.awt.*; // For working with graphics and color
+import java.io.IOException; // For handling input-output exceptions
+import java.io.InputStream; // For handling input streams (e.g., data from HTTP responses)
+import java.net.URI; // For working with URIs (Uniform Resource Identifiers)
+import java.net.http.HttpClient; // For sending HTTP requests
+import java.net.http.HttpRequest; // For creating HTTP requests
+import java.net.http.HttpResponse; // For handling HTTP responses
+
+/**
+ * The AvatarGenerator class is responsible for fetching and displaying randomly generated avatars
+ * from the DiceBear API. It uses HTTP requests to fetch avatar images and displays them in a GUI window.
+ */
 public class AvatarGenerator {
 
+    /**
+     * The main method is the entry point of the program.
+     * It tries to fetch a random avatar and display it using the getRandomAvatarStream and showAvatar methods.
+     */
     public static void main(String[] args) {
 
         try {
+            // Fetches a random avatar as an InputStream
             var avatarStream = AvatarGenerator.getRandomAvatarStream();
+
+            // Displays the fetched avatar in a JFrame window
             AvatarGenerator.showAvatar(avatarStream);
         } catch (IOException | InterruptedException e) {
+            // Handles exceptions that might occur during HTTP requests or InputStream handling
             e.printStackTrace();
         }
```

Comment:
Missing:
- reference types: InputStream, InterruptedException, IOException
- class methods: AvatarGenerator.getRandomAvatarStream, AvatarGenerator.showAvatar
- mention of which package different classes belong to